posts/map_blocks_example.md

```diff
@@ -0,0 +1,191 @@
+---
+author: Matt Long
+date: 2021-4-22
+tags: xarray, dask, pop
+---
+
+# How to Use `xarray.map_blocks` for Vertical Interpolation of a 3D Field
```

Comment:
```suggestion
# How to Use `xarray.map_blocks` to easily parallelize a function that expects xarray objects
```

`map_blocks` is nice and convenient but inefficient.

`map_blocks` takes a block of each dask array, wraps it up in xarray metadata, then wraps all that in to a dataset and sends it to your function. There is a task in the graph for each step for each block. It's a lot of tasks!

 We shouldn't recommend it as a general solution. For fast 1D interpolation, see https://xarray.pydata.org/en/stable/examples/apply_ufunc_vectorize_1d.html

Reply:
Should this example be using `apply_ufunc_1d`?

Reply:
That's what I would do since this function is not too bad to write with plain numpy. But the current form is still a useful demonstration.